onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -307,9 +307,6 @@ void RegisterExecutionProviders(InferenceSession* sess, const std::vector<std::s
     } else if (type == kCudaExecutionProvider) {
 #ifdef USE_CUDA
       RegisterExecutionProvider(sess, *onnxruntime::CreateExecutionProviderFactory_CUDA(cuda_device_id, cuda_mem_limit, arena_extend_strategy));
-      cuda_device_id = 0;
```

Comment:
This is already set at a global level and used in the previous line. There is no need to reset these values again.